registry/spec.md

```diff
@@ -1491,71 +1632,1350 @@ An attempt to delete all versions MUST generate an error.
+| `delivery-count` | `integer` | number of prior unsuccessful delivery attempts |
+
+##### `footer` (AMQP 1.0 Message Footer)
+
+The `footer` property is an object that contains the custom properties of the
+AMQP 1.0 [Message Footer][AMQP 1.0 Message Footer] section.
 
-??? maybe `definitionGroups` ought to be a URL in case the list is large?
+The names of the properties MUST be of type `symbol` and MUST be unique.
+The values of the properties MAY be of any permitted type.
+
+#### "MQTT/3.1.1" and "MQTT/5.0"
+
+The "MQTT/3.1.1" and "MQTT/5.0" formats are used to define messages that are
+sent over [MQTT 3.1.1][MQTT 3.1.1] or [MQTT 5.0][MQTT 5.0] connections. The
+format describes the [MQTT PUBLISH packet][MQTT 5.0] content.
+
+The [`metadata`](#metadata-message-metadata) object contains the elements of the
+MQTT PUBLISH packet directly, with the `user-properties` element corresponding
+to the application properties collection of other protocols.
+
+The following properties are defined. The MQTT 3.1.1 and MQTT 5.0 columns
+indicate whether the property is supported for the respective MQTT version.
+
+| Property | Type | MQTT 3.1.1 | MQTT 5.0 | Description |
+| --- | --- | --- | --- | --- |
+| `qos` | `integer` | yes | yes | Quality of Service level |
+| `retain` | `boolean` | yes | yes | Retain flag |
+| `topic-name` | `string` | yes | yes | Topic name |
+| `payload-format` | `integer` | no | yes | Payload format indicator |
+| `message-expiry-interval` | `integer` | no | yes | Message expiry interval |
+| `response-topic` | `uritemplate` | no | yes | Response topic |
+| `correlation-data` | `binary` | no | yes | Correlation data |
+| `content-type` | `symbol` | no | yes | MIME content type of the payload |
+| `user-properties` | Array | no | yes | User properties |
+
+Like HTTP, the MQTT allows for multiple user properties with the same name,
+so the `user-properties` property is an array of objects, each of which
+contains a single property name and value.
+
+The values of all `string`, `symbol`, and `uritemplate`-typed properties and
+user-properties MAY contain placeholders using the [RFC6570][RFC6570] Level 1
+URI Template syntax. When the same placeholder is used in multiple properties,
+the value of the placeholder is assumed to be identical.
+
+The following example shows a message with the "MQTT/5.0" format, asking for
+QoS 1 delivery, with a topic name of "mytopic", and a user property of
+"my-application-property" with a value of "my-application-property-value":
+
+``` JSON
+{
+  "format" : "MQTT/5.0",
+  "metadata": {
+    "qos": {
+      "value": 1
+    },
+    "retain": {
+      "value": false
+    },
+    "topic-name": {
+      "value": "mytopic"
+    },
+    "user-properties": [
+      { "my-application-property": {
+            "value": "my-application-property-value"
+          }
+      }
+    ]
+  }
+}
+```
+
+#### "Kafka/0.11" format
```

Comment:
Is this really how Kafka identifies the versions?

And why such an old version 0.11? 

Is it because the **KIP** for adding `headers` was added to 0.11.0? (see [KAFKA-4208](https://issues.apache.org/jira/browse/KAFKA-4208)) and since than there was no major update on the records API?